run-session-typechecks hook reports false positive TypeScript errors

## Description

The `run-session-typechecks.ts` hook reports TypeScript errors that don't actually exist when running `pnpm turbo typecheck` or direct `tsc --noEmit`.

## Reproduction

In the constellos/constellos repo (apps/app), the hook reports:

```
apps/app/app/(dashboard)/[spaceSlug]/[projectSlug]/nodes/[nodeType]/[nodeSlug]/page.tsx(19,36): error TS2307: Cannot find module '@/widgets/node-editor' or its corresponding type declarations.
apps/app/app/(dashboard)/[spaceSlug]/[projectSlug]/page.tsx(3,75): error TS2307: Cannot find module '@/shared/ui/card' or its corresponding type declarations.
```

However, these modules exist and resolve correctly:
- `src/widgets/node-editor/index.ts` exports `NodeContentSection`
- `src/shared/ui/card.tsx` exports `Card` components

## Verification

Both of these pass with zero errors:
```bash
cd apps/app && pnpm turbo typecheck  # ✅ passes
cd apps/app && npx tsc --noEmit       # ✅ passes
```

## Expected Behavior

The hook should not report errors for valid TypeScript code that compiles successfully.

## Possible Causes

- Hook may be running tsc from wrong working directory
- Hook may have stale module resolution cache
- Hook may be using different tsconfig or module resolution settings

🤖 Generated with [Claude Code](https://claude.com/claude-code)

--- a/.claude/settings.json
+++ b/.claude/settings.json
@@ -91,18 +91,10 @@
   "enabledPlugins": {
     "plugin-dev@claude-code-plugins": true,
     "feature-dev@claude-code-plugins": true,
-    "github-orchestration@constellos-local": true,
     "nextjs-supabase-ai-sdk-dev@constellos": true,
-    "nextjs-supabase-ai-sdk-dev@constellos-local": true,
-    "project-context@constellos-local": true
+    "github-orchestration@constellos": true
   },
   "extraKnownMarketplaces": {
-    "constellos-local": {
-      "source": {
-        "source": "directory",
-        "path": "${CLAUDE_PROJECT_DIR}/.claude-plugin"
-      }
-    },
     "claude-code-plugins": {
       "source": {
         "source": "github",

--- a/claude-worktree.sh
+++ b/claude-worktree.sh
@@ -766,9 +766,8 @@ _cw_main() {
         # Get worktree-specific cache directory (per-worktree isolation)
         local wt_cache_dir=$(_cw_get_worktree_cache_dir "$worktree_dir")
 
-        # Update symlink to point to this worktree's cache
-        # This ensures each worktree has isolated cache without breaking others
-        _cw_update_cache_symlink "$wt_cache_dir"
+        # NOTE: Symlink update is deferred until AFTER plugins are installed
+        # This prevents race condition where hooks execute from empty cache
 
         local current_path=$(claude plugin marketplace list 2>/dev/null | grep -A1 "constellos-local" | grep "Directory" | sed 's/.*(\(.*\))/\1/')
 
@@ -875,6 +874,12 @@ _cw_main() {
           _cw_item "✘" "${plugin}"
         fi
       done <<< "$plugins"
+    fi
+
+    # Update symlink AFTER plugins are installed to prevent race condition
+    # where SessionStart hooks execute from empty cache directory
+    if [[ -n "$wt_cache_dir" ]]; then
+      _cw_update_cache_symlink "$wt_cache_dir"
     fi
   fi
   set -e

--- a/plugins/nextjs-supabase-ai-sdk-dev/hooks/run-session-typechecks.ts
+++ b/plugins/nextjs-supabase-ai-sdk-dev/hooks/run-session-typechecks.ts
@@ -10,6 +10,7 @@
 import type { StopInput, StopHookOutput } from '../shared/types/types.js';
 import { runHook } from '../shared/hooks/utils/io.js';
 import { findConfigFile } from '../shared/hooks/utils/config-resolver.js';
+import { detectWorktree } from '../shared/hooks/utils/worktree.js';
 import { exec } from 'child_process';
 import { promisify } from 'util';
 
@@ -49,12 +50,22 @@ async function handler(input: StopInput): Promise<StopHookOutput> {
     console.log('[run-session-typechecks] Session ID:', input.session_id);
   }
 
-  // Find tsconfig.json
-  const tsconfigDir = await findConfigFile(input.cwd, 'tsconfig.json');
+  // Normalize cwd to worktree/repo root for consistent config resolution
+  // This fixes false positives in monorepos where input.cwd might be a subdirectory
+  const worktreeInfo = detectWorktree(input.cwd);
+  const searchRoot = worktreeInfo.worktreePath;
+
+  if (DEBUG) {
+    console.log('[run-session-typechecks] Search root:', searchRoot);
+    console.log('[run-session-typechecks] Is worktree:', worktreeInfo.isWorktree);
+  }
+
+  // Find tsconfig.json starting from repo/worktree root
+  const tsconfigDir = await findConfigFile(searchRoot, 'tsconfig.json');
 
   if (!tsconfigDir) {
     // No tsconfig.json found - skip with warning visible in systemMessage
-    const warning = `⚠️ TypeScript check skipped: tsconfig.json not found (searched from ${input.cwd})`;
+    const warning = `⚠️ TypeScript check skipped: tsconfig.json not found (searched from ${searchRoot})`;
     if (DEBUG) {
       console.warn(`[run-session-typechecks] ${warning}`);
     }

--- a/plugins/nextjs-supabase-ai-sdk-dev/hooks/run-task-typechecks.ts
+++ b/plugins/nextjs-supabase-ai-sdk-dev/hooks/run-task-typechecks.ts
@@ -11,6 +11,7 @@ import type { SubagentStopInput, SubagentStopHookOutput } from '../shared/types/
 import { runHook } from '../shared/hooks/utils/io.js';
 import { getAgentEdits } from '../shared/hooks/utils/subagent-state.js';
 import { findConfigFile } from '../shared/hooks/utils/config-resolver.js';
+import { detectWorktree } from '../shared/hooks/utils/worktree.js';
 import { exec } from 'child_process';
 import { promisify } from 'util';
 
@@ -92,12 +93,22 @@ async function handler(input: SubagentStopInput): Promise<SubagentStopHookOutput
     };
   }
 
-  // Find tsconfig.json
-  const tsconfigDir = await findConfigFile(input.cwd, 'tsconfig.json');
+  // Normalize cwd to worktree/repo root for consistent config resolution
+  // This fixes false positives in monorepos where input.cwd might be a subdirectory
+  const worktreeInfo = detectWorktree(input.cwd);
+  const searchRoot = worktreeInfo.worktreePath;
+
+  if (DEBUG) {
+    console.log('[run-task-typechecks] Search root:', searchRoot);
+    console.log('[run-task-typechecks] Is worktree:', worktreeInfo.isWorktree);
+  }
+
+  // Find tsconfig.json starting from repo/worktree root
+  const tsconfigDir = await findConfigFile(searchRoot, 'tsconfig.json');
 
   if (!tsconfigDir) {
     // No tsconfig.json found - skip with warning visible in systemMessage
-    const warning = `⚠️ TypeScript check skipped: tsconfig.json not found (searched from ${input.cwd})`;
+    const warning = `⚠️ TypeScript check skipped: tsconfig.json not found (searched from ${searchRoot})`;
     if (DEBUG) {
       console.warn(`[run-task-typechecks] ${warning}`);
     }

